src/fileloader.h

```diff
@@ -105,6 +109,29 @@ class FileLoader : public QObject
 
         void loadFileImpl(const FileLoadInfo& loadInfo, bool reloaded)
         {
+            // Different codepath for mutating existing data.
+            if (loadInfo.mutateExisting)
+            {
+                std::shared_ptr<GeometryMutator> mutator(new GeometryMutator());
+                if (!mutator->loadFile(loadInfo.filePath))
+                {
+                    g_logger.error("Could not load %s", loadInfo.filePath);
```

Comment:
Perhaps just a TODO comment to make exception handling between `Geometry` and `GeometryMutator` consistent.